### DISCUSSION TITLE
MockBinding example?

### DISCUSSION THREAD

**Discussion by joezappie** [socketio/socket.io#4518] [Q&A]
I'm looking at the documentation for MockBinding but the example is very simple and having a hard time figuring out how to build a more complicated mock binding.

I'm curious, is there a way to create a custom mock binding class that encapsulates all logic, both sending and receiving? Seems like I would have to extend MockPortBinding, but is there a way to tell MockBinding.createPort to use my custom MockPortBinding? 

```
class MyMockBinding extends MockPortBinding {
  constructor(...args) {
    super(...args);
  
    const parser = new ReadlineParser()
    this.port.pipe(parser).on('data', this.onData);
    this.port.on('open', this.onOpen);
  }

  onOpen() {
     setInterval(() => {
       // Stream fake data at 60hz
       const packet = new Packet();
       this.emitData(packet);
     }, 16);
  }

  onData(data) {
      // Real data received
      const request = new Packet(data);
      // Respond to request
      const response = new Packet();
      this.emitData(response );
  }
}

// Then pass the mock binding into create port?
Mockbinding.createPort('/dev/mock', { binding: MyMockBinding });
```

**Comment by darrachequesne** [comment]
Hi! I'm not sure to understand what you are trying to do, could you please explain? Are you trying to use https://serialport.io/docs/api-binding-mock with Socket.IO?
